File: i2f-jdk/i2f-swl/web/i2f-swl/src/i2f-turbo-web/i2f-swl/extension/impl/sm/antherd/SwlAntherdSm3MessageDigester.js
import Sm3Util from "../../../../../i2f-core/crypto/digest/Sm3Util";

/**
 * @return {SwlAntherdSm3MessageDigester}
 * @implements {ISwlMessageDigester}
 */
function SwlAntherdSm3MessageDigester() {

}

/**
 * @param data {String}
 * @return {String}
 */
SwlAntherdSm3MessageDigester.prototype.digest = function (data) {
    return Sm3Util.sign(data)
}
/**
 *
 * @param digest {String}
 * @param data {String}
 * @return {boolean}
 */
SwlAntherdSm3MessageDigester.prototype.verify = function (digest, data) {
    let enc = Sha256Util.sign(data);
    return digest == enc
}

export default SwlAntherdSm3MessageDigester

File: i2f-extension/i2f-extension-swl/src/main/java/i2f/extension/swl/impl/sm/antherd/SwlAntherdSm3MessageDigester.java
package i2f.extension.swl.impl.sm.antherd;

import i2f.extension.jce.sm.antherd.digest.Sm3Digester;
import i2f.swl.std.ISwlMessageDigester;

/**
 * @author Ice2Faith
 * @date 2024/7/10 10:02
 * @desc
 */
public class SwlAntherdSm3MessageDigester implements ISwlMessageDigester {
    private Sm3Digester digester = Sm3Digester.INSTANCE;

    @Override
    public String digest(String data) {
        return digester.digest(data);
    }

    @Override
    public boolean verify(String digest, String data) {
        return digester.verify(digest, data);
    }
}
